sdk/swagger.yaml

```diff
@@ -0,0 +1,720 @@
+      - range
+    PropertyAccessor:
+      allOf:
+        - $ref: "#/components/schemas/Accessor"
+        - type: object
+          properties:
+            value:
+              $ref: "#/components/schemas/Range"
+    Access:
+      type: object
+      properties:
+        receiver:
+          $ref: "#/components/schemas/Range"
+        accessors:
+          type: array
+          items:
+            $ref: "#/components/schemas/Accessor"
+    Interpolation:
+      type: object
+      properties:
+        text:
+          type: string
+        value:
+          type: array
+          items:
+            $ref: "#/components/schemas/PropertyAccessor"
+      required:
+      - text
+    ExprBuiltin:
+      type: object
+      properties:
+        name:
+          type: string
+        nameRange:
+          $ref: "#/components/schemas/Range"
+        argSchema: {}
+        arg:
+          $ref: "#/components/schemas/Expr"
+      required:
+      - name
+    Expr:
+      type: object
+      properties:
+        range:
+          $ref: "#/components/schemas/Range"
+        base:
+          $ref: "#/components/schemas/Expr"
+        schema: {}
+        keyRanges:
+          type: object
+          additionalProperties:
+            $ref: "#/components/schemas/Range"
+        literal: {}
+        interpolate: 
+          type: array
+          items:
+            $ref: "#/components/schemas/Interpolation"
+        symbol:
+          type: array
+          items:
+            $ref: "#/components/schemas/PropertyAccessor"
+        access:
+          type: array
+          items:
+            $ref: "#/components/schemas/Access"
+        list:
+          type: array
+          items:
+            $ref: "#/components/schemas/Expr"
+        object:
+          type: object
+          additionalProperties:
+            $ref: "#/components/schemas/Expr"
+        builtin:
+          $ref: "#/components/schemas/ExprBuiltin"
+    EvaluatedExecutionContext:
+      type: object
+      properties:
+        properties:
+          type: object
+          additionalProperties:
+            $ref: "#/components/schemas/Value"
+        schema: {}
+    Environment:
+      type: object
+      properties:
+        exprs:
+          type: object
+          additionalProperties:
+            $ref: "#/components/schemas/Expr"
+        properties:
+          type: object
+          additionalProperties:
+            $ref: "#/components/schemas/Value"
+        schema: {}
+        executionContext:
+          $ref: "#/components/schemas/EvaluatedExecutionContext"
+    CheckEnvironment:
+      type: object
+      allOf:     
+      - $ref: "#/components/schemas/Environment"
+      - $ref: "#/components/schemas/EnvironmentDiagnostics"
+    Trace:
+      type: object
+      properties:
+        def: 
+          $ref: "#/components/schemas/Range"
+        base:
+          $ref: "#/components/schemas/Value"
+    Value:
+      type: object
+      properties:
+        value: {}

```

Comment:
We could tighten this up, I think, by making it a `oneOf`. The type is `null | boolean | number | string | Value[] | Record<string, Value>`. Not sure how that would play with the code generators, though.

Reply:
The value generators can do one of, but if they have references to other types or circular references they are angry

Reply:
I tried and this cannot do the references to other types because it creates circular references, and the generators will fail.